From da12760e36c7c83db45ab1311ac1c63706953570 Mon Sep 17 00:00:00 2001
From: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
Date: Thu, 14 Apr 2016 23:41:56 +0300
Subject: [PATCH] spi: sh-msiof: fix incorrect udelay

1) fix incorrect udelay caused by overriding HZ constant
udelay uses HZ for constants at compile time.
see arch/arm/include/asm/delay.h

2) speed up polling of CTR register

Signed-off-by: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
---
 drivers/spi/spi-sh-msiof.c | 11 ++++++-------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/drivers/spi/spi-sh-msiof.c b/drivers/spi/spi-sh-msiof.c
index 748119f..faf181a 100644
--- a/drivers/spi/spi-sh-msiof.c
+++ b/drivers/spi/spi-sh-msiof.c
@@ -40,8 +40,7 @@
 
 #include <asm/unaligned.h>
 
-#undef HZ
-#define HZ MAX_SCHEDULE_TIMEOUT
+#define HZ_MSIOF MAX_SCHEDULE_TIMEOUT
 
 struct sh_msiof_chipdata {
 	u16 tx_fifo_size;
@@ -236,11 +235,11 @@ static int sh_msiof_modify_ctr_wait(struct sh_msiof_spi_priv *p,
 	data |= set;
 	sh_msiof_write(p, CTR, data);
 
-	for (k = 100; k > 0; k--) {
+	for (k = 1000; k > 0; k--) {
 		if ((sh_msiof_read(p, CTR) & mask) == set)
 			break;
 
-		udelay(10);
+		udelay(1);
 	}
 
 	return k > 0 ? 0 : -ETIMEDOUT;
@@ -668,7 +667,7 @@ static int sh_msiof_spi_txrx_once(struct sh_msiof_spi_priv *p,
 	}
 
 	/* wait for tx fifo to be emptied / rx fifo to be filled */
-	ret = wait_for_completion_timeout(&p->done, HZ);
+	ret = wait_for_completion_timeout(&p->done, HZ_MSIOF);
 	if (!ret) {
 		dev_err(&p->pdev->dev, "PIO timeout\n");
 		ret = -ETIMEDOUT;
@@ -794,7 +793,7 @@ static int sh_msiof_dma_once(struct sh_msiof_spi_priv *p, const void *tx,
 			sh_msiof_write(p, IER, ier_bits);
 
 			/* wait for tx fifo to be emptied */
-			if (!wait_for_completion_timeout(&p->done, HZ)) {
+			if (!wait_for_completion_timeout(&p->done, HZ_MSIOF)) {
 				dev_err(&p->pdev->dev,
 					"Tx fifo to be emptied timeout\n");
 				ret = -ETIMEDOUT;
-- 
1.9.1

